CONFIG_FS_LOG_LEVEL_OFF option doesn't work with LittleFS

**Describe the bug**
Disabling the logs from LittleFS doesn't work. There's `undefined reference to 'log_const_littlefs'` error during linking.

**To Reproduce**
1. Apply the following patch to disable logs in the LittleFS example:
```
diff --git a/samples/subsys/fs/littlefs/prj.conf b/samples/subsys/fs/littlefs/prj.conf
index 833d700fef..bb46b4cf11 100644
--- a/samples/subsys/fs/littlefs/prj.conf
+++ b/samples/subsys/fs/littlefs/prj.conf
@@ -21,3 +21,5 @@ CONFIG_FLASH_PAGE_LAYOUT=y
 
 CONFIG_FILE_SYSTEM=y
 CONFIG_FILE_SYSTEM_LITTLEFS=y
+
+CONFIG_FS_LOG_LEVEL_OFF=y
```
2. Try to build the `samples/subsys/fs/littlefs/` example.

Please note, that running the example on L475-DISCO requires `CONFIG_MPU_ALLOW_FLASH_WRITE=y` too, but this isn't the problem here.

**Expected behavior**
Successful build, no logs from LittleFS in the runtime.

**Impact**
I implemented config menu over serial port and those logs are annoying.

**Screenshots or console output**
```
kpanek@t470s:~/zephyr-root/zephyr$ west build -b disco_l475_iot1 -t flash samples/subsys/fs/littlefs/
-- west build: running target flash
[1/7] Linking C executable zephyr/zephyr_prebuilt.elf
FAILED: zephyr/zephyr_prebuilt.elf 
: && ccache /home/kpanek/zephyr-sdk-0.11.3/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc    zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj  -o zephyr/zephyr_prebuilt.elf  -Wl,-T  zephyr/linker.cmd  -Wl,-Map=/home/kpanek/zephyr-root/zephyr/build/zephyr/zephyr_prebuilt.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/mpu/libarch__arm__core__aarch32__cortex_m__mpu.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/posix/liblib__posix.a  zephyr/soc/arm/common/cortex_m/libsoc__arm__common__cortex_m.a  zephyr/boards/arm/disco_l475_iot1/libboards__arm__disco_l475_iot1.a  zephyr/subsys/fs/libsubsys__fs.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/flash/libdrivers__flash.a  zephyr/drivers/serial/libdrivers__serial.a  modules/stm32/stm32cube/lib..__modules__hal__stm32__stm32cube.a  modules/littlefs/lib..__modules__fs__littlefs.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj  -L"/home/kpanek/zephyr-sdk-0.11.3/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/thumb/v7e-m/nofp"  -L/home/kpanek/zephyr-root/zephyr/build/zephyr  -lgcc  -Wl,--print-memory-usage  -mcpu=cortex-m4  -mthumb  -mabi=aapcs  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -no-pie  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn && :
Memory region         Used Size  Region Size  %age Used
           FLASH:       57348 B         1 MB      5.47%
            SRAM:        9768 B        96 KB      9.94%
        IDT_LIST:         168 B         2 KB      8.20/home/kpanek/zephyr-sdk-0.11.3/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: modules/littlefs/lib..__modules__fs__littlefs.a(lfs.c.obj): in function `lfs_dir_fetchmatch':
/home/kpanek/zephyr-root/modules/fs/littlefs/lfs.c:778: undefined reference to `log_const_littlefs'
/home/kpanek/zephyr-sdk-0.11.3/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: modules/littlefs/lib..__modules__fs__littlefs.a(lfs.c.obj): in function `lfs_mount':
/home/kpanek/zephyr-root/modules/fs/littlefs/lfs.c:3786: undefined reference to `log_const_littlefs'
/home/kpanek/zephyr-sdk-0.11.3/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: modules/littlefs/lib..__modules__fs__littlefs.a(lfs.c.obj): in function `lfs_alloc':
/home/kpanek/zephyr-root/modules/fs/littlefs/lfs.c:512: undefined reference to `log_const_littlefs'
/home/kpanek/zephyr-sdk-0.11.3/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: modules/littlefs/lib..__modules__fs__littlefs.a(lfs.c.obj): in function `lfs_dir_compact':
/home/kpanek/zephyr-root/modules/fs/littlefs/lfs.c:1717: undefined reference to `log_const_littlefs'
collect2: error: ld returned 1 exit status
%
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /home/kpanek/.local/bin/cmake --build /home/kpanek/zephyr-root/zephyr/build --target flash
```

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - Toolchain: Zephyr SDK
 - Commit SHA or Version used: b8a4b9a1a0dcb3c025b4272ec456fe8547ab13a0

--- a/west.yml
+++ b/west.yml
@@ -117,7 +117,7 @@ manifest:
       revision: 3e9a49d2672ec01435ffbf0d788db6d95ef28de0
     - name: littlefs
       path: modules/fs/littlefs
-      revision: 0aefdda69d236cb01f932117c1f32e9da09c0ec3
+      revision: 9e4498d1c73009acd84bb36036ee5e2869112a6c
     - name: mipi-sys-t
       path: modules/debug/mipi-sys-t
       revision: 957d46bc3ce0d5f628f0d525196bb4db207672ee

